cni/pkg/nodeagent/ztunnelserver.go

```diff
@@ -85,9 +85,22 @@ func (c *connMgr) deleteConn(conn *ZtunnelConnection) {
 	log.Debug("ztunnel disconnected")
 	c.mu.Lock()
 	defer c.mu.Unlock()
-	delete(c.connectionSet, conn)
-	if c.latestConn == conn {
+
+	// Loop over the slice, keeping non-deleted conn pointers
+	// but filtering out the deleted one.
+	var retainedConns []*ZtunnelConnection
+	for _, existingConn := range c.connectionSet {
+		// Not conn-by-pointer that was deleted? Keep it.
+		if existingConn != conn {
+			retainedConns = append(retainedConns, existingConn)
+		}
+	}
+	c.connectionSet = retainedConns
+	newSize := len(c.connectionSet)
+	if newSize == 0 {
 		c.latestConn = nil
+	} else {
+		c.latestConn = c.connectionSet[newSize-1]
```

Comment:
nit: might be easier to just drop `c.latestCon` and make LatestConn() do `c.connectionSet[len(c.connectioSet)-1]`?

Reply:
yep, good call.